kernel/include/ksched.h

```diff
@@ -159,11 +165,17 @@ static inline void unpend_thread_no_timeout(struct k_thread *thread)
 	thread->base.pended_on = NULL;
 }
 
+/*
+ * In a multiprocessor system, z_unpend_first_thread() must lock the scheduler
+ * spinlock _sched_spinlock. However, in a uniprocessor system, that is not
+ * necessary as the caller has already taken precautions (almost always in the
+ * form of locking interrupts).
```

Comment:
Can we add an __ASSERT to validate that this is true in all cases?   Basically just test that interrupts are already locked.  See the code for z_smp_cpu_mobile() for an example (though do **not** include the is_in_isr() test, as ISRs are preemptible!).